cmd/kubeadm/app/cmd/phases/join/kubelet.go

```diff
@@ -206,3 +234,38 @@ func waitForTLSBootstrappedClient() error {
 		return (err == nil), nil
 	})
 }
+
+// printFilesIfDryRunning prints the Static Pod manifests to stdout and informs about the temporary directory to go and lookup
+func printFilesIfDryRunning(data JoinData, joinAsControlPlane bool) error {
```

Comment:
- there is already a function for this in https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/cmd/kubeadm/app/cmd/phases/init/waitcontrolplane.go#L121
can we make that function public and more generic in https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/cmd/kubeadm/app/util/dryrun/dryrun.go instead of adding this new one that is very similar?
- for "join", is "cmd/join/kubelet.go" the right place to print this detail? i assume, yes because KubeletConfigurationFileName is only available then?

Reply:
1. Good point, I have create a new version of printFilesIfDryRunning() under /util/dryrun/dryrun.go that init and join can use, thanks for reminding
2. And the second question, I didn't find a better place to call printFilesIfDryRunning in join, because after runKubeletStartJoinPhase, join process only left NewControlPlaneJoinPhase to run

Reply:
looks like for init the dry-run dumping utility is just called in a random place in waitcontrolplane, ideally it should have been a separate hidden phase (kubeadm supports that).

having the call for join in kubelet-start seems fine for the time being.
maybe one day we can rafactor that.

one benefit here is that that the --dry-run flag cannot be used from phases e.g. `kubeadm init phase ... --dry-run` it can only be used from the main `kubeadm init|join ...`

Reply:
So we agree to leave it here for now? Should I add some todo comment line?

Reply:
OK, done